### ISSUE TITLE
The `promise` dependency silently catches unrelated errors, making it very hard to debug them

### ISSUE THREAD

**Issue by jankeromnes** [espadrine/fsos#3]
In the [janitor](https://github.com/JanitorTechnology/janitor) project, whenever we make a [fleau](https://github.com/espadrine/fleau) template error (e.g. `{{= title in nosuchparser }}`), the error is silently caught by the `promise` dependency module, making it really hard to debug (the server just hangs forever and nothing appears in the console).

It's `try/catch` blocks like [this one](https://github.com/then/promise/blob/1272f80b7f88feb58e09295c43d822a4dcf9609f/src/core.js#L29) that silently catch errors and never report them.

When I hack `./node_modules/fsos/node_modules/promise/lib/core.js` and add `console.erro(ex)` to every `catch` block, I end up seeing the following error for the above problem:

```
Error: Template error: macro "=" didn't work.
Parameters: title,in,nosuchparser
Literal: $_parsers $_parsedtext0 $_parsernames1 $_parserparams2 $_i3 $_scopestr machines projects version title user scripts $_scope $_write $_end
Message: Template error: parser nosuchparser is missing.
    at evalmachine.<anonymous>:74:9
    at ContextifyScript.Script.runInContext (vm.js:32:29)
    at ContextifyScript.Script.runInNewContext (vm.js:38:15)
    at runner (/home/ubuntu/workspace/node_modules/@jankeromnes/camp/node_modules/fleau/fleau.js:247:12)
    at /home/ubuntu/workspace/node_modules/@jankeromnes/camp/lib/camp.js:88:16
    at Array.map (native)
    at ServerResponse.res.template (/home/ubuntu/workspace/node_modules/@jankeromnes/camp/lib/camp.js:84:37)
    at Object.exports.landingPage (/home/ubuntu/workspace/lib/routes.js:105:12)
    at app.route (/home/ubuntu/workspace/app.js:97:12)
    at gotQueries (/home/ubuntu/workspace/node_modules/@jankeromnes/camp/lib/camp.js:887:7)
```

This error shouldn't be silently caught by the `promise` module, as it has nothing to do with it. Silently catching errors is horrible and causes a lot of developer pain. We should either fix it (e.g. by logging all caught errors in `promise`), or switch to using native Promises in Node.js which don't suffer from this problem.

**Comment by jankeromnes** [comment]
See also https://github.com/then/promise/issues/137.

**Comment by jankeromnes** [comment]
Now I got this on Travis for https://github.com/JanitorTechnology/janitor/pull/102:

```
Error: We could not prove email ownership
    at /home/travis/build/JanitorTechnology/janitor/node_modules/email-login/src/api.js:136:30
    at /home/travis/build/JanitorTechnology/janitor/node_modules/email-login/src/registry.js:165:40
    at /home/travis/build/JanitorTechnology/janitor/node_modules/email-login/src/registry.js:108:16
    at tryCallOne (/home/travis/build/JanitorTechnology/janitor/node_modules/fsos/node_modules/promise/lib/core.js:37:12)
    at /home/travis/build/JanitorTechnology/janitor/node_modules/fsos/node_modules/promise/lib/core.js:103:15
    at flush (/home/travis/build/JanitorTechnology/janitor/node_modules/asap/raw.js:50:29)
    at _combinedTickCallback (internal/process/next_tick.js:95:7)
    at process._tickCallback (internal/process/next_tick.js:161:9)
```

Which simply means there was an error, but it's been trapped by `promise`'s silent catch. ☹️ And since it's on Travis, I can't hack `/node_modules/fsos/node_modules/promise/lib/core.js:37` to log the actual error...

**Comment by espadrine** [comment]
We will switch to using native promises, which yield more information when a promise is not handled properly.

**Comment by espadrine** [comment]
Bug considered fixed, published as npm v1.1.2.
